docs/rfc/improving-sync.md

```diff
@@ -0,0 +1,181 @@
+
+4. **Updater**
+
+   Introduced in Store5, an [Updater](https://store.mobilenativefoundation.org/docs/concepts/store5/updater) pushes local changes to the remote source. This is crucial for offline use cases: if a user modifies or deletes data offline, the `Updater` ensures these changes eventually make it to the server once connectivity is restored.
+
+5. **Bookkeeper**
+
+   Another Store5 addition. It tracks local changes that have not yet been successfully synced. Whenever a network call fails, [Bookkeeper](https://store.mobilenativefoundation.org/docs/concepts/store5/bookkeeper) logs the failure so the system can handle it later. Think of it as a ledger that ensures no local changes “fall through the cracks,” including deletions.
+
+6. **Memory Cache**
+
+   In addition to SOT, Store5 typically keeps recently fetched data in memory for quick reads. Deletions must also invalidate this cache to avoid showing stale items.
+
+### Current State of Sync
+
+Store5’s [Bookkeeper](https://store.mobilenativefoundation.org/docs/concepts/store5/bookkeeper) does track failed sync attempts by timestamp, but automatic re-sync is only triggered upon subsequent reads for that key. If a key is written locally and never fetched, it can remain unsynced. The biggest pain point is that there’s no built-in background mechanism to ensure outstanding changes get retried proactively.
+
+> “Values that are written could potentially remain in Store forever if they’re never requested again.”
+>
+> — [#677](https://github.com/MobileNativeFoundation/Store/issues/677)
+
+### Current State of Deletion Support
+
+Prior Store versions (e.g., [Store4](https://youtu.be/raWdIwsDe-g)) introduced fundamental deletion methods (e.g., `clearAll`). However, we still do not have a cohesive end-to-end approach to server-initiated deletions. Without this, local caches may display items that the server already removed. Additionally, we don't have sufficient documentation on how to integrate local or server-side deletions with `Bookkeeper` and `Updater`.
+
+> "One challenge I’m facing is handling deleted data on the server. Since the server doesn’t return deleted items, Store5 doesn’t automatically detect them, which leaves stale records on the client."
+>
+> — [#685](https://github.com/MobileNativeFoundation/Store/discussions/685)
+
+### Recap
+
+We have the building blocks for robust sync ([MutableStore](https://store.mobilenativefoundation.org/docs/concepts/store5/mutable-store), [Bookkeeper](https://store.mobilenativefoundation.org/docs/concepts/store5/bookkeeper), [Updater](https://store.mobilenativefoundation.org/docs/concepts/store5/updater), [SourceOfTruth](https://store.mobilenativefoundation.org/docs/concepts/store5/source-of-truth)). But we need to tighten up deletion support, add background sync scheduling, and clarify conflict resolution flows.
+
+## Considerations
```

Comment:
Another consideration is the ability to retrieve the sync state of items. I wrote this in #677 :

> Related to this, I would like a nice API to retrieve information about failed syncs. This could be used to display information to the user like "5 items have not yet synced to the server"

I'm not sure if Bookkeeper already provides this capability.